sklearn/preprocessing/_data.py

```diff
@@ -941,8 +944,8 @@ def partial_fit(self, X, y=None, sample_weight=None):
                     )
                 # We force the mean and variance to float64 for large arrays
                 # See https://github.com/scikit-learn/scikit-learn/pull/12338
-                self.mean_ = self.mean_.astype(np.float64, copy=False)
-                self.var_ = self.var_.astype(np.float64, copy=False)
+                self.mean_ = self.mean_.astype(xp.float64, copy=False)
+                self.var_ = self.var_.astype(xp.float64, copy=False)
```

Comment:
There is not `.astype` method on the standard `Array`. Instead let's use the `xp.astype` function:

```suggestion
                self.mean_ = xp.astype(self.mean_, xp.float64, copy=False)
                self.var_ = xp.astype(self.var_, xp.float64, copy=False)
```